config/src/test/java/org/springframework/security/config/http/OAuth2LoginBeanDefinitionParserTests.java

```diff
@@ -0,0 +1,739 @@
+		assertThat(authentication.getAuthorities()).first().hasToString("ROLE_USER");
+		assertThat(authentication.getAuthorities()).last().hasToString("ROLE_OIDC_USER");
+	}
+
+	// gh-5488
+	@Test
+	public void successLoginWhenSingleClientRegistrationAndCustomLoginProcessingUrlThenRedirectToDefaultUrlWithCorrectAuthorities()
+			throws Exception {
+		this.spring.configLocations(this.xml("SingleClientRegistration-WithCustomLoginProcessingUrl")).autowire();
+
+		MockHttpSession session = new MockHttpSession();
+		MockHttpServletRequest request = new MockHttpServletRequest("GET", "");
+		request.setServletPath("/login/oauth2/google");
+		request.setSession(session);
+		MockHttpServletResponse response = new MockHttpServletResponse();
+
+		OAuth2AuthorizationRequest authRequest = createOAuth2AuthorizationRequest(
+				clientRegistrationRepository.findByRegistrationId("google-login"));
+		authorizationRequestRepository.saveAuthorizationRequest(authRequest, request, response);
+
+		MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+		params.add("code", "code123");
+		params.add("state", "state123");
+		this.mvc.perform(get("/login/oauth2/google").params(params).session(session))
+				.andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/"));
+
+		Authentication authentication = this.securityContextRepository
+				.loadContext(new HttpRequestResponseHolder(request, response)).getAuthentication();
+		assertThat(authentication.getAuthorities()).hasSize(1);
+		assertThat(authentication.getAuthorities()).first().isInstanceOf(OAuth2UserAuthority.class)
+				.hasToString("ROLE_USER");
+	}
+
+	// gh-5521
+	@Test
+	public void successLoginWhenSingleClientRegistrationAndCustomAuthorizationRequestResolverThenRedirectToCustomAuthURI()
+			throws Exception {
+		this.spring.configLocations(this.xml("SingleClientRegistration-WithCustomAuthorizationRequestResolver"))
+				.autowire();
+
+		this.mvc.perform(get("/oauth2/authorization/google")).andExpect(status().is3xxRedirection())
+				.andExpect(redirectedUrl(MockAuthorizationRequestResolver.REQUEST_URI));
+	}
+
+	// gh-5347
+	@Test
+	public void requestWhenMultipleClientsConfiguredThenRedirectDefaultLoginPage() throws Exception {
+		this.spring.configLocations(this.xml("MultiClientRegistration")).autowire();
+
+		this.mvc.perform(get("/")).andExpect(status().is3xxRedirection())
+				.andExpect(redirectedUrl("http://localhost/login"));
+	}
+
+	@Test
+	public void successLoginWhenSingleClientRegistrationAndCustomLoginPageThenRedirectToCustomLoginPage()
+			throws Exception {
+		this.spring.configLocations(this.xml("SingleClientRegistration-WithCustomLoginPage")).autowire();
+
+		this.mvc.perform(get("/")).andExpect(status().is3xxRedirection())
+				.andExpect(redirectedUrl("http://localhost/custom-login"));
+	}
+
+	// gh-6802
+	@Test
+	public void requestWhenSingleClientRegistrationWithFormLoginPageThenRedirectDefaultLoginPage() throws Exception {
```

Comment:
Please rename to `requestWhenSingleClientRegistrationAndFormLoginConfiguredThenRedirectDefaultLoginPage`